pywinauto/controls/uiawrapper.py

```diff
@@ -490,6 +522,13 @@ def _select(self, item = None):
         else:
             raise IndexError("item not found")
 
+    #-----------------------------------------------------------
+    def is_active(self):
+        """Whether the window is active or not"""
+        minimized = self.iface_window.CurrentWindowVisualState == \
+                       uia_defs.window_visual_state_minimized
+        return (not minimized and self.is_visible() and self.is_enabled())
```

Comment:
As I understand, a window considered "active" only within the given
process. While ```GetForegroundWinfow``` determines a global window between
all the processes. So I think my implementation is good enough to start
with.

On Fri, 22 Jul 2016, 22:36 Vasily Ryabov, <notifications@github.com> wrote:

> In pywinauto/controls/uiawrapper.py
> <https://github.com/pywinauto/pywinauto/pull/213#discussion_r71859511>:
>
> > @@ -490,6 +522,13 @@ def _select(self, item = None):
> >          else:
> >              raise IndexError("item not found")
> >
> > +    #-----------------------------------------------------------
> > +    def is_active(self):
> > +        """Whether the window is active or not"""
> > +        minimized = self.iface_window.CurrentWindowVisualState == \
> > +                       uia_defs.window_visual_state_minimized
> > +        return (not minimized and self.is_visible() and self.is_enabled())
>
> As an initial implementation it solves exception #166
> <https://github.com/pywinauto/pywinauto/issues/166>. But how about the
> case when the app window is normal, enabled and visible but some other app
> opens a window on top of the our app? We can negatively test it with 2
> instances of any MFC sample.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/pywinauto/pywinauto/pull/213/files/604ccfaa4485ec50556eae9dea5c6df2cf857f7a#r71859511>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAwhoIxPKT_84oi5qnfWNmyGBqx29f1Jks5qYJ0QgaJpZM4JSKJN>
> .
>
-- 
Cheers,
Valentin